src/AppsCompiler.ts

```diff
@@ -367,59 +374,63 @@ export class AppsCompiler {
-                        const extendedAppPath = path.isAbsolute(extendedAppShortPath) ? extendedAppShortPath // absolute path
-                            : extendedAppShortPath.startsWith('.')
-                                ? path.join(this.wd, extendedAppShortPath) // relative path
-                                : path.join(this.wd, 'node_modules', extendedAppShortPath); // external path (node_modules)
-                        const engine = import(appsEngineAppPath);
-                        const extendedApp = import(extendedAppPath);
-                        const importedSymbol = renamings.has(extendedSymbol) ? renamings.get(extendedSymbol) : extendedSymbol;
-
-                        extendedApp.then((App) => {
-                            engine.then((engine) => {
-                                const mockInfo = { name: '', requiredApiVersion: '', author: { name: '' } };
-                                const mockLogger = { debug: () => { } };
-                                const extendedApp = new App[importedSymbol](mockInfo, mockLogger);
-
-                                if (!(extendedApp instanceof engine.App)) {
-                                    throw new Error('App must extend apps-engine\'s "App" abstract class.');
-                                }
-                            }).catch(console.error);
-                        });
-                    } catch (err) {
-                        console.error(err, 'Try to run `npm install` in your app folder to fix it.');
-                    }
-                }
-            }
-        });
+        if (!mainClassModule.default && !mainClassModule[mainClassFile]) {
+            throw new Error(`There must be an exported class "${ mainClassFile }" or a default export in the main class file.`);
+        }
 
-        if (!allImports.includes(extendedSymbol)) {
-            throw new Error('App must extend apps-engine\'s "App" abstract class.');
+        const RealApp = mainClassModule.default ? mainClassModule.default : mainClassModule[mainClassFile];
+        const mockInfo = { name: '', requiredApiVersion: '', author: { name: '' } };
+        const mockLogger = { debug: () => { } };
+        const realApp = new RealApp(mockInfo, mockLogger);
+
+        if (!(realApp instanceof EngineBaseApp)) {
+            throw new Error('App must extend apps-engine\'s "App" abstract class.'
+                + ' Maybe you forgot to install dependencies? Try running `npm install`'
+                + ' in your app folder to fix it.',
+            );
         }
     }
 
+    /**
+     * Require a module from the compiled app source files
+     */
+    private requireCompiled(filename: string): any {
+        const exports = {};
+        const context = vm.createContext({
+            require: (filepath: string) => {
+                // Handles Apps-Engine import
+                if (filepath.startsWith('@rocket.chat/apps-engine/definition/')) {
```

Comment:
ah, it seems to be a typo made by me here 🙈

It should be `'@rocket.chat/apps-engine'` right?

Reply:
No, it's actually correct! We don't want apps to require anything that's _out of the definitions folder_. AFAIK this is done in the Apps-Engine just like that. So we're good to go :)

Reply:
ah, it makes sense to me!